### DISCUSSION TITLE
How to pass data to `AsyncError`, when using `AsyncValue.guard`?

### DISCUSSION THREAD

**Discussion by abigotado** [rrousselGit/riverpod#1970] [Q&A]
Hello!

When I use `AsyncValue.guard` if an error occurs, data from the previous state will be lost. Is there a way to pass data from the previous state to `AsyncError` using the guard?

I've tried to do like this inside my `AsyncNotifier`:
```dart
final AsyncValue<PaginationState<T>> previousState = state;
state = await AsyncValue.guard(someAsyncMethod);
if (state.hasError) state = state.copyWithPrevious(previousState);
```
But that doesn't work with `ref.listen`, as it reacts to the first error state, which doesn't have data yet.

**Comment by rrousselGit** [comment]
There's an issue about it https://github.com/rrousselGit/riverpod/issues/1862

**Reply by abigotado** [reply]
I've seen it. I think it is concerning `.when` operator and its `skipError` argument. And my question is more about the the case of using `.hasError` operator. In this case it's not possible to use `skipError`.
